projects/packages/search/src/class-module-control.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * Jetpack Search: Module_Control class
+ *
+ * @package    jetpack-search
+ */
+
+namespace Automattic\Jetpack\Search;
+
+use Automattic\Jetpack\Status;
+use Jetpack_Options;
+
+/**
+ * To get and set Searh module settings
+ */
+class Module_Control {
+	/**
+	 * We use the same options as Jetpack the plugin to flag whether Search is active.
+	 */
+	const JETPACK_ACTIVE_MODULES_OPTION_KEY       = 'active_modules';
+	const JETPACK_SEARCH_MODULE_SLUG              = 'search';
+	const SEARCH_MODULE_INSTANT_SEARCH_OPTION_KEY = 'instant_search_enabled';
+
+	/**
+	 * Singleton Instance
+	 *
+	 * @var Module_Control
+	 */
+	protected static $instance;
+
+	/**
+	 * Hide contruct
+	 */
+	protected function __construct() {
+	}
+
+	/**
+	 * Return the singleton
+	 */
+	public static function get_instance() {
+		if ( is_null( static::$instance ) ) {
+			static::$instance = new static();
+		}
+		return static::$instance;
+	}
+
+	/**
+	 * Returns a boolean for whether of the module is enabled.
+	 *
+	 * @return bool
+	 */
+	public function is_active() {
+		return in_array( self::JETPACK_SEARCH_MODULE_SLUG, self::get_active_modules(), true );
+	}
+
+	/**
+	 * Returns a boolean for whether instant search is enabled.
+	 *
+	 * @return bool
+	 */
+	public function is_instant_search_enabled() {
+		return (bool) get_option( self::SEARCH_MODULE_INSTANT_SEARCH_OPTION_KEY );
+	}
+
+	/**
+	 * Activiate Search module
+	 */
+	public function activate() {
+		/**
+		 * Fires before a module is activated.
+		 *
+		 * @since 2.6.0
+		 *
+		 * @param string $module Module slug.
+		 * @param bool $exit Should we exit after the module has been activated. Default to true.
+		 * @param bool $redirect Should the user be redirected after module activation? Default to true.
+		 */
+		do_action( 'jetpack_pre_activate_module', self::JETPACK_SEARCH_MODULE_SLUG );
```

Comment:
The reason I'm still using the same filters is for capability - there could be code that relies on these filters. When it runs inside Jetpack the plugin, we'd need the exactly the same behaviour as other parts. And when running standalone, we could possibly use other names, but still better to be consistent I think.

Reply:
Using the same filter names makes sense to me, but I think in the future we could potentially extract this out of the Jetpack plugin and into a package, so we do not have to duplicate things.

Although in this specific case, extracting this out of the Jetpack class may be more trouble than it's worth, I'm not sure it would be really more readable.